src/Transport/Config/ExtensionPoints/AzureServiceBusNamespacePartitioningSettings.cs

```diff
@@ -29,7 +29,7 @@ public AzureServiceBusNamespacePartitioningSettings(SettingsHolder settings)
             return this;
         }
 
-        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString)
+        public AzureServiceBusNamespacePartitioningSettings AddNamespace(string name, string connectionString, NamespacePurpose purpose = NamespacePurpose.Partitioning)
```

Comment:
> I don't think that is a good idea, UseNamespaceNameInsteadOfConnectionString is only temporary, in future releases it will be default. then we'd have to change the API again

Agree, but we don't know when `securing connection string` will become the default. For sure not in the next version, I think. 
With this context in mind, we have to provide the best configuration API for the current version (IMO). When we'll upgrade to a newer version we'll think about how to change them.